pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -545,9 +549,17 @@ func (m *managerImpl) evictPod(pod *v1.Pod, gracePeriodOverride int64, evictMsg
 	// If the pod is marked as critical and static, and support for critical pod annotations is enabled,
 	// do not evict such pods. Static pods are not re-admitted after evictions.
 	// https://github.com/kubernetes/kubernetes/issues/40573 has more details.
-	if kubelettypes.IsCriticalPod(pod) && kubepod.IsStaticPod(pod) {
-		klog.Errorf("eviction manager: cannot evict a critical static pod %s", format.Pod(pod))
-		return false
+	if kubepod.IsStaticPod(pod) {
+		isCriticalPod := false
+		if mirrorPod, ok := m.mirrorPodFunc(pod); ok && mirrorPod != nil {
+			isCriticalPod = kubelettypes.IsCriticalPod(mirrorPod)
+		} else {
+			isCriticalPod = kubelettypes.IsCriticalPod(pod)
```

Comment:
do we still need this case?  The description made it sound like we _only_ needed to check the mirror pod.

Reply:
@dashpole Theoretically yes. The `else` block here is just to catch some exceptional case such like a static pod doesn't have a corresponding mirror pod  in memory.

Reply:
IIUC, static pods will never be critical?  Or is that possible if they are using the critical pod annotation, rather than priority?

Reply:
In most cases, static pods are critical.

Critical pod annotation is another story, it's now deprecated in favor of `PriorityClass`. You can think of it as another style of "system-node-critical" priority.

Reply:
Right, i'm asking if the reason why we care about the "exceptional case" is because some static pods _could_ be critical (even if we don't have a mirror pod).

If we don't have a mirror pod in memory, the right thing to do might just be to skip it (don't evict it).  If we mistakenly evict a critical static pod, that can have drastic consequences for nodes.

Reply:
I get your point. I will go logging an error instead of proceeding on this case, and when we hit this case, it means we're hitting an in-memory data inconsistency. Sounds good?

Reply:
sounds good!